test/io/test_io.py

```diff
@@ -1341,3 +1341,25 @@ def test_passes_with_3_sec_statement_and_4_sec_statement_timeout(defaultenv):
         response = postgrest.session.post("/rpc/four_sec_timeout")
 
         assert response.status_code == 204
+
+
+def test_db_error_logging_to_stderr(defaultenv, metapostgrest):
+    "verify that DB errors are logged to stderr"
+
+    role = "timeout_authenticator"
+    set_statement_timeout(metapostgrest, role, 1000)
+
+    env = {
+        **defaultenv,
+        "PGUSER": role,
+        "PGRST_DB_ANON_ROLE": role,
+    }
+
+    with run(env=env) as postgrest:
+        response = postgrest.session.get("/rpc/sleep?seconds=1.5")
+        assert response.status_code == 500
+
+        # ensure the message appears on the logs
+        output = sorted(postgrest.read_stdout(nlines=2))
```

Comment:
It's a bit confusing how this says `read_stdout` and the test description says `stderr`. I've seen `read_stdout` is only used in a couple of places. Perhaps we can refactor it? 

Not sure why stderr is redirected to STDOUT here:

https://github.com/PostgREST/postgrest/blob/3a6c33da1e40c554c774898b9aea73c9e17abde8/test/io/postgrest.py#L106-L112

https://github.com/PostgREST/postgrest/blob/3a6c33da1e40c554c774898b9aea73c9e17abde8/test/io/postgrest.py#L60-L70

Reply:
> Perhaps we can rename it so it clearly says `read_stderr`?

Yeah, agree. It looks like it's used only for errors.

> Not sure why stderr is redirected to STDOUT here:

According [to the docs](https://docs.python.org/3/library/subprocess.html#frequently-used-arguments):

> If capture_output is true, stdout and stderr will be captured. When used, the internal [Popen](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) object is automatically created with stdout=PIPE and stderr=PIPE. The stdout and stderr arguments may not be supplied at the same time as capture_output. If you wish to capture and combine both streams into one, use stdout=PIPE and stderr=STDOUT instead of capture_output.

So, it may have been configured for convenience? Perhaps, there may be some caveats we're missing.

Reply:
Another option would be  `read_stdouterr` and maybe a comment to the function

Reply:
Maybe leave it for another PR with a `refactor: ..` commit.

Reply:
Yes, was about to say the same.